cdk-testsuite/tests/openshiftSanity.py

```diff
@@ -35,39 +37,120 @@ def setUp(self):
            
     def test_python_project(self):
         '''
-        TBD
+        Runs sanity on openshift s2i python source
+            1. login
+            2. new project create
+            3. App deploy using the registry from config file
+            4. Validation for successful deployment
+            5. Routing
+            6. Pod status
+            7. project delete
+            8. logout
+        Args:
+            self (object): Object of the current method
+            openshift_URL (string) : ip and port of openshift server to be used
+            openshift_USERNAME (string): username of openshift web console to be used
+            openshift_PASSWORD (string): password of openshift web console to be used
+            openshift_python_PROJECT (string): name of the python project to be added to the openshift server
+            openshift_python_REGISTRY (string): python registry path/location
+            service_python_NAME (string): name of the python service to be exposed outside
         '''
         openshift.new_project(self, self.params.get('openshift_URL'), self.params.get('openshift_USERNAME'), 
                               self.params.get('openshift_PASSWORD'), self.params.get('openshift_python_PROJECT'), 
                               self.params.get('openshift_python_REGISTRY'), self.params.get('service_python_NAME'))
     
     def test_ruby_project(self):
         '''
-        TBD
+        Runs sanity on openshift s2i ruby source
+            1. login
+            2. new project create
+            3. App deploy using the registry from config file
+            4. Validation for successful deployment
+            5. Routing
+            6. Pod status
+            7. project delete
+            8. logout
+        Args:
+            self (object): Object of the current method
+            openshift_URL (string) : ip and port of openshift server to be used
+            openshift_USERNAME (string): username of openshift web console to be used
+            openshift_PASSWORD (string): password of openshift web console to be used
+            openshift_ruby_PROJECT (string): name of the ruby project to be added to the openshift server
+            openshift_ruby_REGISTRY (string): ruby registry path/location
+            service_ruby_NAME (string): name of the ruby service to be exposed outside
         '''
         openshift.new_project(self, self.params.get('openshift_URL'), self.params.get('openshift_USERNAME'), 
```

Comment:
Yes its good to have test validation in test case file but here in our scenario if we do that it only increase the lines of code with repetition in test case file . So i believe to make a library function where we can pass the args and do validation.